src/json-helper.c

```diff
@@ -96,18 +96,26 @@ gchar* json_get_string(JsonNode *json_node, const gchar *path, GError **error)
         return g_steal_pointer(&res_str);
 }
 
-gint64 json_get_int(JsonNode *json_node, const gchar *path)
+gint64 json_get_int(JsonNode *json_node, const gchar *path, GError **error)
 {
-        gint64 res_int = -1;
-        JsonNode *result = json_path_query(path, json_node, NULL);
-        if (result) {
-                JsonArray *results = json_node_get_array(result);
-                if (json_array_get_length( results ) > 0) {
-                        res_int = json_array_get_int_element(results, 0);
-                }
+        g_autoptr(JsonNode) result = NULL;
+
+        g_return_val_if_fail(error == NULL || *error == NULL, -1);
```

Comment:
glib functions return a `0` to indicate an error. I'd like to stay consistent with this.

I do not really like the violation of common glib error handling that we do not have a return code clearly indicating an error, but glib functions do it the same way and it would otherwise require an output argument, thus ok.

Maybe we should add a note in the signature documentation that only a `0` with GError set is an error indicator?

Reply:
Force-pushed 0 return on error and mentioned it in the doc string.